### ISSUE TITLE
no-use-before-define does not report bindings in nested blocks

### ISSUE THREAD

**Issue by jrvidal** [eslint/eslint#2960]
Example:
```js
{
    alert(x);
    let x = 1;
}
```

This code does not trigger any errors from `no-use-before-define` and it definitely should.

#2877 might influence the discussion about this issue.
